### PR TITLE
#164: Fixed default values for click parameters of type multiple=true

### LINKED ISSUE #164: Export API throws exception when force_rebuild_from not specified
# Background

- force_rebuild_from has the default None
- However, at some point it is expected to be an iterable
- We need to initialize it as empty tuple, if it is not set

# Acceptance Criteria

- Fix for the bug
- Test such that it doesn't happen again

### PR DESCRIPTION
**Pull request by tomuben**
fixes #164 

RCA: There is an inconsistency in click, when using parameter of type multiple=true. The runtime object of the click command, indicates a default-value=None; however, when invoking the command without the respective parameter, click passes an empty tuple.

Solution: 
1. Implement a special handling for the test when the default value of a parameter is None, but the parameter is of type multiple=true. 
2. Convert all type hints of kind `Optional[Tuple[....]]` in click commands to `Tuple[...]` and replace default values `None` => `tuple()`

### CODE REVIEW

File: test/test_click_api_consistency.py [hunk 1]
```diff
@@ -17,19 +17,25 @@ def is_click_command(obj: Any) -> bool:
...
     @staticmethod
-    def _replace_list_with_tuple(x: Any):
+    def _adjust_default_value_for_multiple(x: Any):
         """
         Click stores default values as list if 'Multiple'=true. However, for plain Python methods we
         need to use (immutable) tuples for declaring default value. Hence, we need to convert lists with tuples
         for comparison.
+        Also, default values for none-required parameters of with multiple=true are marked as "None"
+        in the command object, but in reality click invokes the command with "tuple()" for that parameter.
+        (Bug in click????)
```
Comment (redcatbear) [inline test/test_click_api_consistency.py:27]: If you are not sure, write check that project's bug tracker. Write a ticket please, if the project has no related ticket.

**Review by redcatbear — APPROVED** [review]

File: exasol_script_languages_container_tool/cli/commands/run_db_tests.py [hunk 2]
```diff
@@ -106,7 +106,7 @@ def run_db_test(flavor_path: Tuple[str, ...],
                 reuse_test_container: bool,
                 reuse_test_environment: bool,
                 force_rebuild: bool,
-                force_rebuild_from: Optional[Tuple[str, ...]],
+                force_rebuild_from: Tuple[str, ...],
                 force_pull: bool,
```
Comment (tkilias) [inline exasol_script_languages_container_tool/cli/commands/run_db_tests.py:110]: Default is missing

File: exasol_script_languages_container_tool/lib/api/run_db_tests.py [hunk 3]
```diff
@@ -13,11 +13,11 @@
 
 def run_db_test(flavor_path: Tuple[str, ...],
                 release_goal: Tuple[str, ...] = ('release',),
-                generic_language_test: Optional[Tuple[str, ...]] = None ,
-                test_folder: Optional[Tuple[str, ...]] = None,
-                test_file: Optional[Tuple[str, ...]] = None,
+                generic_language_test: Tuple[str, ...] = tuple(),
+                test_folder: Tuple[str, ...] = tuple(),
+                test_file: Tuple[str, ...] = tuple(),
                 test_language: Tuple[str, ...] = (None,),
```
Comment (tkilias) [inline exasol_script_languages_container_tool/lib/api/run_db_tests.py:19]: That is wierd?

**Review by tkilias — CHANGES_REQUESTED** [review]
In general, I would suggest also test which actually call tbe API, but we can do this maybe in another PR.

File: exasol_script_languages_container_tool/lib/api/run_db_tests.py (hunk 3 above)
Comment (tomuben) [reply exasol_script_languages_container_tool/lib/api/run_db_tests.py:19]: corresponding click default value is `[None]`:
see https://github.com/exasol/script-languages-container-tool/blob/1ee272b47194e6251de662036abfbe945bfe3835/exasol_script_languages_container_tool/cli/commands/run_db_tests.py#L37

File: test/test_click_api_consistency.py (hunk 1 above)
Comment (tomuben) [reply test/test_click_api_consistency.py:27]: On one hand I agree with you that we should report bugs. But on the other hand I the test uses some internal structures from click, which is not what normal clients are supposed to do. I don't believe it's worth the effort to open, track and respond the ticket.

**Review by tkilias — APPROVED** [review]

**Review by redcatbear — APPROVED** [review]

**Pull request merged**